src/Voron/Data/BTrees/Tree.Stream.cs

```diff
@@ -87,60 +79,54 @@ public void Init(Tree parent, Slice key, Slice? tag, int? initialNumberOfPagesPe
 
             public void Write(Stream stream)
             {
-                _localBuffer = ArrayPool<byte>.Shared.Rent(512 * Constants.Size.Kilobyte);
```

Comment:
wouldnt it be better to do:
```
using (_parent._tx.Allocator.Allocate(512 * Constants.Size.Kilobyte, out Span<byte> localBuffer))
```

then you would only change this line and remove `finally` at the end?

Reply:
I would go with that one.

Reply:
Looks good.
I decrease the scope of the buffer to increase the chance of quick release of the buffer by just moving the pointer.
The buffer is needed only in the read section.
But I guess you prefer minimal changes in the code.

Reply:
yeah

Reply:
everything that goes to lower branch (in this case 5.4) goes into higher branches (6.2, 7.0) via regular merges that I'm doing